services/registry/docker-compose.yml

```diff
@@ -66,7 +66,51 @@ services:
         mode: 0644
       - source: registry_config.yml
         target: /etc/docker/registry/config.yml
-
+  registry-pull-through-cache:
+    hostname: "registry-ptc-{{.Node.Hostname}}-{{.Task.Slot}}"
+    image: registry:2.8.2
+    environment:
+      REGISTRY_HTTP_SECRET: ${REGISTRY_PULL_THROUGH_CACHE_HTTP_SECRET}
+      # S3
+      REGISTRY_STORAGE: s3
+      REGISTRY_STORAGE_S3_ACCESSKEY: ${REGISTRY_S3_ACCESS_KEY}
+      REGISTRY_STORAGE_S3_SECRETKEY: ${REGISTRY_S3_SECRET_KEY}
+      REGISTRY_STORAGE_S3_REGIONENDPOINT: "${REGISTRY_S3_ENDPOINT}"
+      REGISTRY_STORAGE_S3_SECURE: "true"
+      REGISTRY_STORAGE_S3_BUCKET: ${REGISTRY_PULL_THROUGH_CACHE_S3_BUCKET}
+      REGISTRY_STORAGE_S3_REGION: ${REGISTRY_AWS_REGION}
+      REGISTRY_STORAGE_CACHE_BLOBDESCRIPTOR: inmemory
```

Comment:
this is for the pull cache right? are there other options in the end?
if we scale the registry then there will be multiple caches right? thus increasing the number of pulls?

Reply:
also this can potentially become large? is the underlying EC2 able to sustain the space?

Reply:
Removed this ENV altogether